Bulldozer.CSV/HelperMethods.cs

```diff
@@ -0,0 +1,502 @@
+                                 && (
+                                    a.Location.Street1 == personAddressImport.Street1
+                                    && a.Location.Street2 == personAddressImport.Street2
+                                    && a.Location.City == personAddressImport.City
+                                    && a.Location.County == personAddressImport.County
+                                    && a.Location.State == personAddressImport.State
+                                    && a.Location.Country == personAddressImport.Country
+                                    && a.Location.PostalCode == personAddressImport.PostalCode
+                                 ) ).Any();
+
+                            if ( !addressAlreadyExistsExactMatch )
+                            {
+                                var locationService = new LocationService( rockContextForPersonUpdate );
+
+                                Location location = locationService.Get( personAddressImport.Street1, personAddressImport.Street2, personAddressImport.City, personAddressImport.State, personAddressImport.PostalCode, personAddressImport.Country, false );
+
+                                if ( !primaryFamilyGroupLocations.Where( a => a.GroupLocationTypeValueId == personAddressImport.GroupLocationTypeValueId && a.LocationId == location.Id ).Any() )
+                                {
+                                    var groupLocation = new GroupLocation();
+                                    groupLocation.GroupId = primaryFamily.Id;
+                                    groupLocation.GroupLocationTypeValueId = personAddressImport.GroupLocationTypeValueId;
+                                    groupLocation.IsMailingLocation = personAddressImport.IsMailingLocation;
+                                    groupLocation.IsMappedLocation = personAddressImport.IsMappedLocation;
+
+                                    if ( location.GeoPoint == null && personAddressImport.Latitude.HasValue && personAddressImport.Longitude.HasValue )
+                                    {
+                                        location.SetLocationPointFromLatLong( personAddressImport.Latitude.Value, personAddressImport.Longitude.Value );
+                                    }
+
+                                    groupLocation.LocationId = location.Id;
+                                    groupLocationService.Add( groupLocation );
+
+                                    addressesUpdated = true;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if ( personAttributesUpdated )
+                {
+                    person.SaveAttributeValues();
+                }
+
+                var updatedRecords = rockContextForPersonUpdate.SaveChanges( true );
+
+                return addressesUpdated || personAttributesUpdated || updatedRecords > 0;
```

Comment:
Address and Attribute bools in return are not necessary, as the .SaveChanges above already save them. Maybe this is for a dictionary?